### PR TITLE
Collect `TypedStructField` tokens in the language server.

### PR DESCRIPTION
**Pull request by JoshuaBatty**
This PR allows us to collect both the name and the type as tokens for fields in Struct and Storage Declarations. This is working for most types apart from arrays and tuples as outlined in #2486. 

Go to definition support is now also working for fields in Struct expressions.

### CODE REVIEW

File: sway-lsp/src/server.rs [hunk 1]
```diff
@@ -291,25 +291,25 @@ mod tests {
...
             .unwrap()
             .parent()
             .unwrap()
             .join("test/src/e2e_vm_tests/test_programs/should_pass/language")
-            .join("enum_if_let")
+            .join("struct_field_access")
     }
 
-    fn sway_example_dir() -> PathBuf {
+    fn _sway_example_dir() -> PathBuf {
```
Comment (sezna) [inline sway-lsp/src/server.rs:303]: what's this for?

**Review by sezna — DISMISSED** [review]

File: sway-lsp/src/server.rs (hunk 1 above)
Comment (JoshuaBatty) [reply sway-lsp/src/server.rs:303]: I have 2 helper methods for testing against the sway examples folder and another for the e2e language folder. Clippy complains if there are any unused functions so I underscore the one I'm not currently using for development. 

This is actually one of the main motivations for me wanting to inspect the syntax tree in the editor for the currently open file in VScode.

Comment (sezna) [reply sway-lsp/src/server.rs:303]: Ah ok, fair. what about just doing `#[allow(dead_code)]` on both?

**Review by sezna — APPROVED** [review]

File: sway-lsp/src/server.rs (hunk 1 above)
Comment (JoshuaBatty) [reply sway-lsp/src/server.rs:303]: Yeah good idea. I'll add that to my next PR. Cheers.

**Review by mohammadfawaz — APPROVED** [review]
Compiler changes seem reasonable to me!

**Pull request merged**